DSCResources/MSFT_xDnsServerForwarder/MSFT_xDnsServerForwarder.schema.mof

```diff
@@ -0,0 +1,7 @@
+[ClassVersion("1.0.0"), FriendlyName("xDnsServerForwarder")]
+class MSFT_xDnsServerForwarder :  OMI_BaseResource
+{
+    [Key] String IPAddress;
+    [Write] Boolean RemoveAll;
```

Comment:
I understand why you did it, but I was just thinking If IPAddress was an array, to Remove all the forwarders you'd just set the array parameter to be an empty array (or not provide the parameter at all):
```powershell
xDnsServerForwarder RemoveAllForwardersfromDNSServer1 {
   IsSingleInstance = 'Yes'
   IPAddress = @()
}
```
That way the config never needs to know the current state of the resource (as you say) - it always replaces the current values if different.

Just my 2c :smile: